### PR TITLE
Cache ptex and scie-jump downloads.

### PR DESCRIPTION
**Pull request by jsirois**

### CODE REVIEW

**Comment by sureshjoshi** [comment]
What's the use case where this cache is used? After a successful download, but an interrupted extraction?

**Comment by sureshjoshi** [comment]
I ask because I thought the extracted digest effectively acts as an unpacked result that future ptex/jump calls refer to?

**Comment by jsirois** [comment]
@sureshjoshi exactly right. These components are used to build a scie and need to be as-advertised. I can only blame this on trying code during layovers 🤦

**Comment by sureshjoshi** [comment]
Okay, and it looks like it's not co-located with the `nce`, correct?

`macOS.../Library/Caches/scie-pants/dev/downloads/...`?

**Comment by jsirois** [comment]
Yeah, this is not a scie app, it's the build process for one using it's own bespoke cache of downloaded ptex and scie-jump binaries to build the scie with.

**Comment by jsirois** [comment]
Basically I need to get around to working on a-scie/lift, that blank repo is intended to be a high level tool (shipped as a scie, written in Python) that is what normal people use to build scie apps. It would handle getting scie-jump and ptex for you to build the scie with and lots of code here would go away.

**Comment by kaos** [comment]
so many word-puns 😂

**Comment by sureshjoshi** [comment]
👍🏽  Just confirming my current understanding

**Review by sureshjoshi — APPROVED** [review]

**Comment by jsirois** [comment]
@sureshjoshi PTAL at the last commit - things were buggy in the way you suggest before that!

File: package/src/main.rs [hunk 1]
```diff
@@ -444,33 +444,27 @@ fn fetch_a_scie_project(
...
-    if 0 == std::fs::metadata(&cached_binary)
-        .map_err(|e| {
-            Code::FAILURE.with_message(format!(
-                "Failed to retrieve metadata for {path}: {e}",
-                path = cached_binary.display()
-            ))
-        })?
-        .len()
-    {
+    if !target_dir.exists() {
```
Comment (sureshjoshi) [inline package/src/main.rs:462]: What's was the atomicity failure vs the previous commits? I thought based on the previous commits that the `target_dir` is created before this?

**Comment by jsirois** [comment]
Noting after this that the only thing slow about the dev cycle is the `tools.pex` build step. That we can't use Pants for (bootstrapping circularity here! There is no Pants launcher yet, we're building it!); so I'll need to look into some bespoke caching for that step as a follow up.

File: package/src/main.rs (hunk 1 above)
Comment (sureshjoshi) [reply package/src/main.rs:462]: >  the partial binary would be there

Ahhh, okay, I thought this process was also atomic. That makes sense now.

**Review by sureshjoshi — APPROVED** [review]

File: package/src/main.rs (hunk 1 above)
Comment (sureshjoshi) [reply package/src/main.rs:462]: My original thought was not about corruption, but rather just network caching - I didn't think we could half unpack a digest

Comment (jsirois) [reply package/src/main.rs:462]: Aha, ok.

**Pull request merged**